test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+        validate_and_cleanup_ext_list(bms, 0);
+    }
+
+    SECTION("Allocate one ext blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_cleanup_ext_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple ext blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_cleanup_ext_list(bms, 3);
+    }
+}
+
+TEST_CASE("Block session: __block_size_alloc", "[block_session]")
+{
+    WT_SIZE *sz;
+
+    sz = nullptr;
+    REQUIRE(__ut_block_size_alloc(nullptr, &sz) == 0);
+    validate_and_cleanup_size_block(sz);
+}
+
+TEST_CASE("Block session: __block_size_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_cleanup_size_list(bms, 0);
+    }
+
+    SECTION("Allocate one size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_cleanup_size_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_cleanup_size_list(bms, 3);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate with null block manager session and no extent cache")
+    {
+        std::shared_ptr<MockSession> session_test_bm = MockSession::buildTestMockSession();
+        WT_EXT *ext;
+
+        REQUIRE(__wti_block_ext_alloc(session_test_bm->getWtSessionImpl(), &ext) == 0);
+        validate_and_cleanup_ext_block(ext);
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with fake zero cache extent count")
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        // Construct extent cache with one item.
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 0;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(cached_ext == ext);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with one extent in cache ")
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        // Construct extent cache with one item.
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 1;
```

Comment:
Here we should fill ext->next with junk and ensure it is cleared.

Reply:
I don't quite get what you mean here, the second allocation is meant to work as a cache mechanism. Therefore the ext->next[0] shouldn't be cleared in this place. what were you looking for? @luke-pearson

Reply:
I'm referring to testing this code branch: 
```
        /* Clear any left-over references. */
        for (i = 0; i < ext->depth; ++i)
            ext->next[i] = ext->next[i + ext->depth] = NULL;
```

I think we could test that?

Reply:
Oh yes, i see what you mean, I have added junk next now.